### PR TITLE
Remove neighbour as an interpolation method

### PR DESCRIPTION
**Pull request by nabobalis**
Fixes https://github.com/sunpy/sunpy/issues/6531

### CODE REVIEW

**Comment by dstansby** [comment]
Looks like there's some (more?) tests that need removing

**Comment by nabobalis** [comment]
Yeah I missed those since I searched for the UK spelling and forgot about American English.

File: changelog/6537.removal.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+:meth:`sunpy.image.resample` will no longer accept "neighbour" as an interpolation method.
```
Comment (wtbarnes) [inline changelog/6537.removal.rst:1]:
```suggestion
:meth:`sunpy.image.resample` no longer accepts "neighbour" as an interpolation method.
```

**Review by wtbarnes — APPROVED** [review]

File: changelog/6537.removal.rst [hunk 2]
```diff
@@ -0,0 +1 @@
+:meth:`sunpy.image.resample` no longer accepts "neighbour" as an interpolation method.
```
Comment (dstansby) [inline changelog/6537.removal.rst:1]:
```suggestion
:meth:`sunpy.image.resample` no longer accepts "neighbour" as an interpolation method.
Use "nearest" instead.
```

**Review by dstansby — APPROVED** [review]

**Pull request merged**